docs/commands/rpc-protocol.md

```diff
@@ -2840,30 +2839,61 @@ Clear work peers node list until restart
 ### work_validate 
 Check whether **work** is valid for block  
 
+!!! warning "Semantics change in V21.0"
+    In V21.0, when the optional **difficulty** is *not* given, **valid** is not included in the response.
+
+    Use the new response fields **"valid_all"** and **"valid_receive"** taking into account the subtype of the block using this work value.
+
 **Request:**
 ```json
 {
   "action": "work_validate",
   "work": "2bf29ef00786a6bc",
-  "hash": "718CC2121C3E641059BC1C2CFC45666C99E8AE922F7A807B7D07B62C995D79E2",
-  "difficulty": "ffffffd21c3933f3"
+  "hash": "718CC2121C3E641059BC1C2CFC45666C99E8AE922F7A807B7D07B62C995D79E2"
 }
 ```  
-**Response:**
+**Response since v21.0:**
 ```json
 {
-  "valid": "1",
-  "difficulty": "ffffffd21c3933f4",
-  "multiplier": "1.394647"
+  "valid_all": "1",
+  "valid_receive": "1",
+  "difficulty": "fffffff93c41ec94",
+  "multiplier": "1.182623871097636" // calculated from the default base difficulty
 }
 ```
 
-*Since version 19.0+:* The response also includes the work `value` in hexadecimal format, and a `multiplier` from the base difficulty (not from the optionally given difficulty).
+??? abstract "Response up to v20.0"
+    ```json
+    {
+      "valid": "1",
+      "difficulty": "fffffff93c41ec94", // since v19.0
+      "multiplier": "9.4609" // since v19.0
+    }
+    ```
 
 **Optional "difficulty"**
 
 _version 19.0+_  
-Difficulty value (16 hexadecimal digits string, 64 bit). Uses **difficulty** value to validate work  
+Difficulty value (16 hexadecimal digits string, 64 bit). Uses **difficulty** value to validate work. Defaults to the network base difficulty.  
+
+**Request with given "difficulty"**  
```

Comment:
Request here doesn't include difficulty option

Reply:
Shouldn't `valid` in the response be `1` given the difficulty passed into the example?

Reply:
The given difficulty is the maximum, no work could be valid. The thought was this would show best that `valid` is based on the request difficulty

Reply:
Ah, I read that backwards as 0 difficulty. So then why are `valid_all` and `valid_receive` = `1` - shouldn't those be `0` as well?

Reply:
No, because those don't care about the difficulty in the request, only the network difficulties (normal and receive)

Reply:
Ah yes. Thanks for clarifying.